server/src/test/java/org/elasticsearch/cluster/routing/allocation/FilterRoutingTests.java

```diff
@@ -92,14 +93,13 @@ public void testIndexFilters() {
 
         logger.info("Building initial routing table");
 
-        MetaData initialMetaData = MetaData.builder()
-                .put(IndexMetaData.builder("test").settings(settings(Version.CURRENT)
-                        .put("index.number_of_shards", 2)
-                        .put("index.number_of_replicas", 1)
-                        .put("index.routing.allocation.include.tag1", "value1,value2")
-                        .put("index.routing.allocation.exclude.tag1", "value3,value4")
-                        .build()))
-                .build();
+        final Settings.Builder settingsBuilder = settings(Version.CURRENT).put("index.number_of_shards", 2).put("index.number_of_replicas", 1);
+        if (randomBoolean()) {
```

Comment:
I like what you're doing here to improve these tests. However, I have been moving away from using randomness as a poor man's way to get branch coverage. I think that randomness is good for testing edge cases by randomly varying input data, but I don't like seeing it to half of the time cover one branch and half of the time cover another. I would prefer to see two tests here.